src/cart/crt_init.c

```diff
@@ -290,7 +290,8 @@ crt_plugin_fini(void)
 
 	for (i = 0; i < CRT_SRV_CONTEXT_NUM; i++) {
 		D_FREE(crt_plugin_gdata.cpg_prog_cbs[i]);
-		D_FREE(crt_plugin_gdata.cpg_prog_cbs_old[i]);
+		if (crt_plugin_gdata.cpg_prog_cbs_old[i])
+			D_FREE(crt_plugin_gdata.cpg_prog_cbs_old[i]);
```

Comment:
Curious why this is necessary? I.e. I thought we were generally removing explicit checks like this elsewhere

Reply:
We are, and in fact this may be one that I've changed myself (I have a pre-commit hook to do this normally), but in this instance the D_FREE() call is made many hundreds of times with a NULL parameter, so it adds noise to the log files and time to process all the files.